CONTRIBUTING.md

```diff
@@ -0,0 +1,122 @@
+# Contributing
+
+Contributions are welcome, and they are greatly appreciated! Every little bit
+helps, and credit will always be given.
+
+You can contribute in many ways:
+
+## Types of Contributions
+
+### Report Bugs
+
+Report bugs at https://github.com/GNS-Science/nzshm-common-py/issues.
+
+If you are reporting a bug, please include:
+
+- Your operating system name and version.
+- Any details about your local setup that might be helpful in troubleshooting.
+- Detailed steps to reproduce the bug.
+
+### Fix Bugs
+
+Look through the GitHub issues for bugs. Anything tagged with "bug" and "help
+wanted" is open to whoever wants to implement it.
+
+### Implement Features
+
+Look through the GitHub issues for features. Anything tagged with "enhancement"
+and "help wanted" is open to whoever wants to implement it.
+
+### Write Documentation
+
+nzhsm-common-py could always use more documentation, whether as part of the
+official nzhsm-common-py docs, in docstrings, or even on the web in blog posts,
+articles, and such.
+
+### Submit Feedback
+
+The best way to send feedback is to file an issue at https://github.com/GNS-Science/nzhsm-common-py/issues.
+
+If you are proposing a feature:
+
+- Explain in detail how it would work.
+- Keep the scope as narrow as possible, to make it easier to implement.
+- Remember that this is a volunteer-driven project, and that contributions
+  are welcome :)
+
+## Get Started!
+
+Ready to contribute? Here's how to set up `nzhsm-common-py` for local development.
+
+1. Fork the `nzhsm-common-py` repo on GitHub.
+2. Clone your fork locally
+
+   ```
+   $ git clone git@github.com:your_name_here/nzhsm-common-py.git
+   ```
+
+3. Ensure [poetry](https://python-poetry.org/docs/) is installed.
+4. Install dependencies and start your virtualenv:
+
+   ```
+   $ poetry install -E test -E doc -E dev
```

Comment:
Some of the `test_geometry.py` tests fail in this configuration.
It might be worth going for `poetry install --all-extras` instead.

Reply:
@DrCuriosity  agreed  !